src/TestFramework/Factory.php

```diff
@@ -105,19 +106,33 @@ public function create(string $adapterName, bool $skipCoverage): TestFrameworkAd
             );
         }
 
-        if ($adapterName === TestFrameworkTypes::CODECEPTION) {
-            $codeceptionConfigPath = $this->configLocator->locate(TestFrameworkTypes::CODECEPTION);
+        $extensionsClassExists = class_exists('\Infection\ExtensionInstaller\GeneratedExtensionsConfig');
 
-            return CodeceptionAdapterFactory::create(
-                (new TestFrameworkFinder('codecept'))->find(),
-                $this->tmpDir,
-                $codeceptionConfigPath,
-                null,
-                $this->jUnitFilePath,
-                $this->projectDir,
-                $this->infectionConfig->getSourceDirectories(),
-                $skipCoverage
-            );
+        if ($extensionsClassExists) {
+            $installedExtensions = \Infection\ExtensionInstaller\GeneratedExtensionsConfig::EXTENSIONS;
+
+            foreach ($installedExtensions as $packageName => $installedExtension) {
+                $factory = $installedExtension['extra']['class'];
```

Comment:
if it's from a generated file why not providing the class directly instead?

Reply:
It's a convention to use `extra.infection` object to store additional data, and for now it's only a `class`.

Later, when we add additional extension points like Mutators, custom OutputFormatters etc, this can be not only a `class`. 

So, `extension-installer` just copies all the information from the plugin, and it's a responsibility of the client code to decide how to use it